props/blaster_BC_buttons.h

```diff
@@ -0,0 +1,861 @@
+    SaberBase::SetLockup(SaberBase::LOCKUP_NONE);
+      SetNextActionF(NEXT_ACTION_BLOW, len);
+  }
+// Volume Menu
+    void VolumeMenu() {
+    if (!mode_volume_) {
+      mode_volume_ = true;
+      if (SFX_vmbegin) {
+        sound_library_.SayEnterVolumeMenu();
+      } else {
+        beeper.Beep(0.1, 1000);
+        beeper.Beep(0.1, 2000);
+        beeper.Beep(0.1, 3000);
+      }
+      STDOUT.println("Enter Volume Menu");
+      SaberBase::DoEffect(EFFECT_VOLUME_LEVEL, 0);
+    } else {
+      mode_volume_ = false;
+      if (SFX_vmend) {
+        sound_library_.SayVolumeMenuEnd();
+      } else {
+        beeper.Beep(0.1, 3000);
+        beeper.Beep(0.1, 2000);
+        beeper.Beep(0.1, 1000);
+      }
+      STDOUT.println("Exit Volume Menu");
+    }
+  }
+
+  void VolumeUp() {
+    SaberBase::DoEffect(EFFECT_VOLUME_LEVEL, 0);
+    if (dynamic_mixer.get_volume() < VOLUME) {
+      dynamic_mixer.set_volume(std::min<int>(VOLUME + VOLUME * 0.1,
+        dynamic_mixer.get_volume() + VOLUME * 0.10));
+      if (SFX_volup) {
+        hybrid_font.PlayPolyphonic(&SFX_volup);
+      } else {
+        beeper.Beep(0.10, 2000);
+        beeper.Beep(0.20, 2500);
+      }
+      STDOUT.print("Volume Up - Current Volume: ");
+      STDOUT.println(dynamic_mixer.get_volume());
+    } else {
+      QuickMaxVolume();
+    }
+  }
+
+  void VolumeDown() {
+    SaberBase::DoEffect(EFFECT_VOLUME_LEVEL, 0);
+    if (dynamic_mixer.get_volume() > (0.10 * VOLUME)) {
+      dynamic_mixer.set_volume(std::max<int>(VOLUME * 0.1,
+        dynamic_mixer.get_volume() - VOLUME * 0.10));
+      if (SFX_voldown) {
+        hybrid_font.PlayPolyphonic(&SFX_voldown);
+      } else {
+        beeper.Beep(0.10, 2000);
+        beeper.Beep(0.20, 1500);
+      }
+      STDOUT.print("Volume Down - Current Volume: ");
+      STDOUT.println(dynamic_mixer.get_volume());
+    } else {
+      QuickMinVolume();
+    }
+  }
+
+  void QuickMaxVolume() {
+    SaberBase::DoEffect(EFFECT_VOLUME_LEVEL, 0);
+    dynamic_mixer.set_volume(VOLUME);
+    if (SFX_volmax) {
+      hybrid_font.PlayPolyphonic(&SFX_volmax);
+    } else {
+      beeper.Beep(0.5, 3000);
+    }
+    STDOUT.print("Maximum Volume \n");
+  }
+
+  void QuickMinVolume() {
+    SaberBase::DoEffect(EFFECT_VOLUME_LEVEL, 0);
+    dynamic_mixer.set_volume(std::max<int>(VOLUME * 0.1,
+    dynamic_mixer.get_volume() - VOLUME * 0.90));
+    if (SFX_volmin) {
+      hybrid_font.PlayPolyphonic(&SFX_volmin);
+      } else {
+        beeper.Beep(0.5, 1000);
+      }
+      STDOUT.print("Minimum Volume \n");
+  }
+
+  // Make clash do nothing except unjam if jammed.
+  void Clash(bool stab, float strength) override {
+    if (!mode_volume_) {
+      if (is_jammed_) {
+        is_jammed_ = false;
+        SaberBase::DoEffect(EFFECT_UNJAM, 0);
+      }
+    }
+  }
+
+  // Make swings do nothing
+  void DoMotion(const Vec3& motion, bool clear) override {
+    PropBase::DoMotion(Vec3(0), clear);
+  }
+
+RefPtr<BufferedWavPlayer> wav_player;
```

Comment:
variables should generally go at the bottom (and be indented)

Reply:
I put it above Event2 where I saw it in other props, and
I just added more bools to the spot that blaster.h has them.
So should the RefPtr (and all the other declarations) be moved to the bottom of the file?

Reply:
Yes.

Reply:
Working on it.

Reply:
done